src/sentry/static/sentry/app/views/releases/list/projectList.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import {tct, t} from 'app/locale';
+import ProjectBadge from 'app/components/idBadge/projectBadge';
+import {Project} from 'app/types';
+import space from 'app/styles/space';
+import GlobalSelectionLink from 'app/components/globalSelectionLink';
+import Hovercard, {Body as HoverCardBody} from 'app/components/hovercard';
+
+const MAX_PROJECTS_IN_HOVERCARD = 5;
+
+type Props = {
+  projects: Project[];
+  orgId: string;
+  version: string;
+  maxLines?: number;
+};
+
+const ProjectList = ({projects, orgId, version, maxLines = 2}: Props) => {
+  let visibleProjects: Project[], hiddenProjects: Project[];
+
+  if (projects.length <= maxLines) {
+    visibleProjects = projects;
+    hiddenProjects = [];
+  } else {
+    // because we need one line for `and X more`
+    visibleProjects = projects.slice(0, maxLines - 1);
+    hiddenProjects = projects.slice(maxLines - 1, projects.length);
+  }
+
+  const hoverCardHead = t('Release Projects');
+  const hovercardBody = (
+    <HovercardContentWrapper>
+      <ProjectList
+        projects={hiddenProjects}
+        orgId={orgId}
+        version={version}
+        maxLines={MAX_PROJECTS_IN_HOVERCARD}
+      />
+    </HovercardContentWrapper>
+  );
+
+  return (
+    <React.Fragment>
+      {visibleProjects.map(project => (
+        <StyledProjectBadge project={project} avatarSize={14} key={project.slug} />
+      ))}
+      {hiddenProjects.length > 0 && (
+        <StyledHovercard
+          header={hoverCardHead}
+          body={hovercardBody}
+          show={hiddenProjects.length <= MAX_PROJECTS_IN_HOVERCARD ? undefined : false}
```

Comment:
This could be true/false. I don't think you need undefined.

Reply:
I do, because if hovercard gets a show prop, it respects that and not hover actions (false means never show up, true means be always open). So basically what this is doing is telling hovercard to never show up if there are too many projects, otherwise open on hover.